src/attributes.jl

```diff
@@ -461,6 +461,14 @@ struct RawParameter <: AbstractOptimizerAttribute
     name::Any
 end
 
+"""
+    NumberThreads()
+
+An optimizer attribute for setting the number of threads used for an
+optimization. Values are integers and the default value is 1.
```

Comment:
1 isn't the default for all solvers. I think many commercial solvers are multi-threaded by default. Is this intended to change the solver default?

Reply:
It was intended to change the solver default. I think we should have a default value so a model can be optimized with different solver with the same number of threads without any work. With that in mind 1 seemed the only choice.

Reply:
We say: If `NumberOfThreads` is set to `nothing`, then it is unspecified (solver default or solver-specific attribute) and if it is set to an `Integer` then it overwrites any solver-specific attribute.

Reply:
> It was intended to change the solver default.

I don't support this. It's likely to lead to support questions like "why is JuMP slower than X?". 

> so model can be optimized with different solver with the same number of threads without any work

It's not much work at all to set `NumberOfThreads`.

+1 for `nothing` as the default value, which has no effect on the solver defaults. If `NumberOfThreads` is set to an integer then it overrides the default.

Reply:
> I don't support this. It's likely to lead to support questions like "why is JuMP slower than X?".

Ok, I change that